p2p/discovery/message.go

```diff
@@ -18,39 +20,189 @@ const (
 	NeighborsMsg MsgType = 5
 )
 
-type Ping struct {
+const (
+	VERSION uint = 1
+)
+
+type ping struct {
 	Version uint // TODO add version check
-	ID      NodeID
+	SelfID  NodeID
+
+	to NodeID
```

Comment:
coding style:
SelfID: uppercase
to: lowercase.

Reply:
this is supposed